### PR TITLE
DDF-5714 Add admin configurable icon definitions

### PR DESCRIPTION
**Pull request by Bdthomson**
#### What does this PR do?
Adds an admin config to customize icons in the UI.

#### Who is reviewing it? 

@vinamartin 
@jlcsmith 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@djblue

#### How should this be tested?

Change an admin configured icon for a datatype and verify that it changes in the UI.

#### What are the relevant tickets?
Fixes: #5714 

#### Screenshots

image

image
_Example: PNG image type uses the gamepad icon as defined in the admin UI_

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/config/ConfigurationApplication.java [hunk 1]
```diff
@@ -833,6 +836,24 @@ private void setProxyForTerrain(Map terrainProvider) {
...
 
+  public void setIconConfig(List<String> icons) {
+    Map<String, Map<String, String>> iconMap = new HashMap<>();
+
+    icons.forEach(
+        icon -> {
+          String key = icon.split("=")[0];
+          String[] config = icon.split("=")[1].split(",");
+          Map<String, String> tempMap = new HashMap<>();
+          tempMap.put("className", config[0]);
```
Comment (bdeining) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/config/ConfigurationApplication.java]: should we be checking the size of `config` after the split?

Comment (Bdthomson) [reply catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/config/ConfigurationApplication.java]: I can add that

Comment (Bdthomson) [reply catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/config/ConfigurationApplication.java]: https://github.com/codice/ddf/pull/5715/commits/700e08948477f64fb6261225cea6847283d2ceaa Good enough?

**Comment by Bdthomson** [comment]
build now

**Comment by Bdthomson** [comment]
build now

**Comment by jlcsmith** [comment]
build now

**Comment by Bdthomson** [comment]
Closing in favor of https://github.com/codice/ddf-ui/pull/65

**Pull request closed without merging**